enumerable.rb

```diff
@@ -0,0 +1,179 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = self if instance_of?(Array)
+    arr = to_a if instance_of?(Range)
+    arr = flatten if instance_of?(Hash)
+
+    count = 0
+    while count < arr.length
+      yield(arr[count])
+      count += 1
+    end
+    arr
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    arr = self if instance_of?(Array)
+    arr = to_a if instance_of?(Range)
+    arr = flatten if instance_of?(Hash)
+
+    count = 0
+    while count < arr.length
+      yield(arr[count], count)
+      count += 1
+    end
+    arr
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_arr = []
+    my_each { |it| new_arr << it if yield(it) }
+    new_arr
+  end
+
+  def my_all?(param = nil)
+    if block_given?
+      my_each { |item| return false if yield(item) == false }
+      return true
+    elsif param.nil?
+      my_each { |num| return false if num.nil? || num == false }
+    elsif !param.nil? && (param.is_a? Class)
+      my_each { |num| return false if num.class != param }
+    elsif !param.nil? && param.instance_of?(Regexp)
+      my_each { |_num| return false unless arg.match(n) }
+    else
+      my_each { |_num| return false if n != param }
+    end
+    true
+  end
+
+  def my_any?(param = nil)
```

Comment:
enumerables #my_any? returns false if the block ever returns a value false or nil
```
      Failure/Error: expect(array.my_any?(&false_block)).to eq(array.any?(&false_block))
      
        expected: false
             got: true
```

enumerables #my_any? when no block or argument is given returns false if at least one of the collection is not true
```
      Failure/Error: expect(false_array.my_any?).to be false_array.any?
      
        expected false
             got true
```

enumerables #my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer
```
      Failure/Error: expect(words.my_any?(Integer)).to be words.any?(Integer)
      
        expected false
             got true
```

enumerables #my_any? when a Regex is passed as an argument returns true if any of the collection matches the Regex
```
      Failure/Error: expect(words.my_any?(/d/)).to be words.any?(/d/)
```

enumerables #my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
```
      Failure/Error: expect(words.my_any?(/z/)).to be words.any?(/z/)
```

enumerables #my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
```
      Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')
```

enumerables #my_any? when a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern
```
      Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')
```

# 
### `my_any?` should be identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))